src/core/net/dhcp6_client.cpp

```diff
@@ -275,12 +275,23 @@ otError Dhcp6Client::Solicit(uint16_t aRloc16)
     SuccessOrExit(error = AppendIaAddress(*message, aRloc16));
     SuccessOrExit(error = AppendRapidCommit(*message));
 
+#if OPENTHREAD_ENABLE_DHCP6_MULTICAST_SOLICIT
+    messageInfo.GetPeerAddr().mFields.m16[0] = HostSwap16(0xff03);
+    messageInfo.GetPeerAddr().mFields.m16[1] = HostSwap16(0x0000);
```

Comment:
nit: probably unnecessary to explicitly zero out these fields - the [MessageInfo constructor](https://github.com/openthread/openthread/blob/80f982301ca199edfacdd8176f037822a3d144fc/src/core/net/socket.hpp#L62) zeros these fields out.